examples/clone.py

```diff
@@ -1,18 +1,37 @@
-#!/usr/bin/python
-# This trivial script demonstrates how to clone a remote repository.
-#
-# Example usage:
-#  python examples/clone.py git://github.com/jelmer/dulwich dulwich-clone
+"""Clone.
+
+This trivial script demonstrates how to clone a remote repository.
+
+Example usage:
+    2. python examples/clone.py git://github.com/jelmer/dulwich.git dulwich
+    1. python examples/clone.py git://github.com/jelmer/dulwich.git
+"""
+
 
 import sys
+
+from os.path import basename
+
 from getopt import getopt
+
 from dulwich import porcelain
 
-opts, args = getopt(sys.argv, "", [])
-opts = dict(opts)
+
+_, args = getopt(sys.argv, "", [])
+
 
 if len(args) < 2:
     print("usage: %s host:path path" % (args[0], ))
     sys.exit(1)
 
-porcelain.clone(args[1], args[2])
+elif len(args) < 3:
+    repository = basename(args[1].split(":")[-1])
+    if repository[-4:] == ".git":
```

Comment:
Rather than modifying the args, perhaps just assign to a variable with the local repository name and have logic to set that variable if it wasn't provided?

Reply:
This seems to be the case, it can be changed and streamlined.